dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -358,15 +358,20 @@ public String getContent(String mainDescriptorPath, String mainDescriptor, Set<S
                         stepToType.put(workflowStepId, expressionToolType);
                     } else if (run instanceof Map) {
                         // must be import or include
-                        Object importVal = ((Map)run).get("import");
+                        Object importVal = ((Map)run).containsKey("$import") ? ((Map)run).get("$import") : ((Map)run).get("import");
                         if (importVal != null) {
                             secondaryFile = importVal.toString();
                         }
 
-                        Object includeVal = ((Map)run).get("include");
+                        Object includeVal = ((Map)run).containsKey("$include") ? ((Map)run).get("$include") : ((Map)run).get("include");
                         if (includeVal != null) {
                             secondaryFile = includeVal.toString();
                         }
+
+                        if (secondaryFile == null) {
```

Comment:
I don't understand the whole flow, but this is saying that if `run` is a Map, then return an Optional.empty if the map doesn't reference an import. Are we saying we always want that? I guess so, and I'm saying in English what the code does, and this may be the whole point of the fix, but it also seems like a bit of a change, so just making sure.

Reply:
According to the comment there, if the run is a Map then it must be an import or include. If I don't return the optional.empty then the code will fail later at line 396 since the secondary file is never set which means `stepToType` won't get that  `workflowStepId`. That was causing the NPE before. It's possible there are other options other than import or include if it's a map though, but I don't really know CWL.